Source/Parallelization/WarpXComm.cpp

```diff
@@ -1038,24 +1064,39 @@ void WarpX::AddCurrentFromFineLevelandSumBoundary (
-                ng_depos_J += bilinear_filter.stencil_length_each_dir-1;
-                ng_depos_J.min(ng);
-                MultiFab jfc(J_cp[lev+1][idim]->boxArray(),
-                             J_cp[lev+1][idim]->DistributionMap(), J_cp[lev+1][idim]->nComp(), ng);
-                bilinear_filter.ApplyStencil(jfc, *J_cp[lev+1][idim], lev+1);
-
-                // buffer patch of fine level
-                MultiFab jfb(current_buf[lev+1][idim]->boxArray(),
-                             current_buf[lev+1][idim]->DistributionMap(), current_buf[lev+1][idim]->nComp(), ng);
-                bilinear_filter.ApplyStencil(jfb, *current_buf[lev+1][idim], lev+1);
-
-                MultiFab::Add(jfb, jfc, 0, 0, current_buf[lev+1][idim]->nComp(), ng);
-                ablastr::utils::communication::ParallelAdd(mf, jfb, 0, 0, current_buf[lev + 1][idim]->nComp(),
-                                                           ng, IntVect::TheZeroVector(), WarpX::do_single_precision_comms, period);
-
-                WarpXSumGuardCells(*J_cp[lev+1][idim], jfc, period, ng_depos_J, 0, J_cp[lev+1][idim]->nComp());
+                //// coarse patch of fine level
+                //ng += bilinear_filter.stencil_length_each_dir-1;
+                //ng_depos_J += bilinear_filter.stencil_length_each_dir-1;
+                //ng_depos_J.min(ng);
+                //MultiFab jfc(J_cp[lev+1][idim]->boxArray(),
+                //             J_cp[lev+1][idim]->DistributionMap(), J_cp[lev+1][idim]->nComp(), ng);
+                //bilinear_filter.ApplyStencil(jfc, *J_cp[lev+1][idim], lev+1);
+
+                //// buffer patch of fine level
+                //MultiFab jfb(current_buf[lev+1][idim]->boxArray(),
+                //             current_buf[lev+1][idim]->DistributionMap(), current_buf[lev+1][idim]->nComp(), ng);
+                //bilinear_filter.ApplyStencil(jfb, *current_buf[lev+1][idim], lev+1);
+
+                //MultiFab::Add(jfb, jfc, 0, 0, current_buf[lev+1][idim]->nComp(), ng);
+                //ablastr::utils::communication::ParallelAdd(mf, jfb, 0, 0, current_buf[lev + 1][idim]->nComp(),
+                //                                           ng, IntVect::TheZeroVector(), WarpX::do_single_precision_comms, period);
+
+                //WarpXSumGuardCells(*J_cp[lev+1][idim], jfc, period, ng_depos_J, 0, J_cp[lev+1][idim]->nComp());
+
+                ApplyFilterJ(J_cp, lev+1);
+                ApplyFilterJ(current_buf, lev+1);
+
+                MultiFab::Add(
+                    *current_buf[lev+1][idim], *J_cp[lev+1][idim],
+                    0, 0, current_buf[lev+1][idim]->nComp(), ng);
+
+                ablastr::utils::communication::ParallelAdd(
+                    mf, *current_buf[lev+1][idim], 0, 0,
+                    current_buf[lev+1][idim]->nComp(),
+                    ng, amrex::IntVect(0),
+                    do_single_precision_comms, period);
+
+                SumBoundaryJ(J_cp, lev+1, period);
```

Comment:
@WeiqunZhang

When you have some time, would you be able to tell why calling the new functions `ApplyFilterJ` and `SumBoundaryJ` as done here is not equivalent to the code commented right above? Thank you.

P.S. That the two codes are not equivalent (and, in fact, that the new code is wrong) is inferred by the fact that the MR tests using filtering and current buffers fail (e.g., the test `Langmuir_multi_2d_MR_psatd`).